core/src/main/java/org/ehcache/EhcacheManager.java

```diff
@@ -242,8 +245,16 @@ public void release() {
 
     final CacheLoaderWriterFactory cacheLoaderWriterFactory = serviceLocator.findService(CacheLoaderWriterFactory.class);
     final CacheLoaderWriter<? super K, V> loaderWriter;
+    CacheLoaderWriter<? super K, V> decorator = null;
     if(cacheLoaderWriterFactory != null) {
       loaderWriter = cacheLoaderWriterFactory.createCacheLoaderWriter(alias, config);
+      CacheLoaderWriterConfiguration cacheLoaderWriterConfiguration = ServiceLocator.findSingletonAmongst(CacheLoaderWriterConfiguration.class, config.getServiceConfigurations().toArray());
+      if(cacheLoaderWriterConfiguration != null && cacheLoaderWriterConfiguration.isWriteBehind()) {
+        final WriteBehindDecoratorLoaderWriterProvider factory = serviceLocator.findService(WriteBehindDecoratorLoaderWriterProvider.class);
+        decorator = factory.createWriteBehindDecoratorLoaderWriter((CacheLoaderWriter<K, V>)loaderWriter, cacheLoaderWriterConfiguration); 
```

Comment:
Can we avoid type casting of the decorator? Shouldn't decorator also have same type as the CacheLoaderWriter?